cleverpush/src/main/java/com/cleverpush/service/NotificationDataProcessor.java

```diff
@@ -108,9 +111,11 @@ public static void process(Context context, Notification notification, Subscript
                 notifications = new LimitedSizeQueue<>();
             }
             notifications.setCapacity(maximumNotifications);
-
             if (notification.getCreatedAt() == null || notification.getCreatedAt().equalsIgnoreCase("")) {
-                String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.US).format(new Date());
+                String currentDate = null;
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
```

Comment:
Will this prevent a crash? And can we still define `currentDate` to any value for Android < 7 ?

Reply:
@jprajapati1996

Reply:
```
            if (notification.getCreatedAt() == null || notification.getCreatedAt().equalsIgnoreCase("")) {
                String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX", Locale.US).format(new Date());
                notification.setCreatedAt(currentDate);
            }

```